ArduCopter/mode.h

```diff
@@ -232,6 +232,67 @@ class Mode {
     };
     static AutoYaw auto_yaw;
 
+#if PRECISION_LANDING == ENABLED
+    // Precision Landing State Machine
+    class PrecLand_StateMachine {
+    public:
+        // Initialize the state machine. This is called everytime we switch to a mode that can do prec landing
+        void init();
+
+        // Run Prec Land State Machine. During Prec Landing, we might encouter four scenarios:
+        // 1. We had the target in sight, but have lost it now. 2. We never had the target in sight and user wants to land.
+        // 3. We have the target in sight and can continue landing. 4. The sensor is out of range
+        // This method deals with all of these scenarios
+        void update_precland_state_machine();
+
+    private:
+
+        // Target is lost (i.e we had it in sight some time back), this method helps decide on what needs to be done next
+        // The chosen action depends on user set landing strictness
+        void target_lost_actions();
```

Comment:
```suggestion
        void do_target_lost_actions();
```

(verb+noun rather than just noun)

(I reckon we should canonicalise in our code on `run` or `do` BTW :-) )